### PR TITLE
Updates timer and converts mob nests to use the timer system

### PR DESCRIPTION
**Pull request by printer16**
## About The Pull Request

Updates the timer system from upstream TG and changes the mob nests to use a timer system instead of checking for mobs in range when it can't even fire.

## Why It's Good For The Game

Includes some fixes with the timer system and might help with some lag from mob nests.

## Pre-Merge Checklist
- [X] Your Pull Request contains no breaking changes
- [X] You tested your changes locally, and they work.
- [X] There are no new Runtimes that appear.
- [X] You documented all of your changes.

## Changelog
:cl:
/:cl:

### CODE REVIEW

**Comment by printer16** [comment]
I suggest test merging this to see it's effect.

File: code/__HELPERS/_lists.dm [hunk 1]
```diff
@@ -50,7 +50,7 @@
 			var/__BIN_LEFT = 1;\
 			var/__BIN_RIGHT = __BIN_CTTL;\
 			var/__BIN_MID = (__BIN_LEFT + __BIN_RIGHT) >> 1;\
-			var/##TYPECONT/__BIN_ITEM;\
+			var ##TYPECONT/__BIN_ITEM;\
```
Comment (hecksadecimal) [inline code/__HELPERS/_lists.dm:53]: Was this intended? Doesn't look right. Maybe I'm missing something though.

Comment (printer16) [reply code/__HELPERS/_lists.dm:53]: Yes appears to be a by-product of https://github.com/tgstation/tgstation/pull/53217

Comment (hecksadecimal) [reply code/__HELPERS/_lists.dm:53]: Kind of weird but alright, seems fine then.

**Review by hecksadecimal — APPROVED** [review]

**Comment by hecksadecimal** [comment]
> 
> 
> I suggest test merging this to see its effect.

I'll set this up for a testmerge.

**Comment by printer16** [comment]
> > I suggest test merging this to see its effect.
> 
> I'll set this up for a testmerge.

Looks good after several rounds with no issues from what I can see. I don't have access to the stats, so I'm not sure the exact performance gain here.

**Comment by hecksadecimal** [comment]
> 
> 
> > > I suggest test merging this to see its effect.
> > 
> > 
> > I'll set this up for a testmerge.
> 
> Looks good after several rounds with no issues from what I can see. I don't have access to the stats, so I'm not sure the exact performance gain here.

Between this and the quirk removal, TD seems to be sitting at an average of 21% during peak hours.
However, I've not heard of any crashes happening. So that's pretty good.

**Pull request merged**